var/spack/repos/builtin/packages/ccache/package.py

```diff
@@ -53,14 +55,24 @@ class Ccache(CMakePackage):
 
     depends_on("zstd", when="@4.0:")
 
-    depends_on("hiredis@0.13.3:", when="@4.4:")
+    depends_on("hiredis@0.13.3:", when="@4.4: +redis")
     depends_on("pkgconfig", type="build", when="@4.4:")
 
     conflicts("%gcc@:5", when="@4.4:")
     conflicts("%clang@:4", when="@4.4:")
 
     def cmake_args(self):
-        return [self.define("ENABLE_TESTING", False)]
+        spec = self.spec
+        args = [
+            self.define("ENABLE_TESTING", False),
+            self.define("ENABLE_DOCUMENTATION", False),
+            self.define_from_variant("REDIS_STORAGE_BACKEND", "redis"),
+            self.define("ZSTD_FROM_INTERNET", False),
+        ]
+        if spec.satisfies("+redis"):
+            args.append(self.define("HIREDIS_FROM_INTERNET", False))
```

Comment:
Uh, this should remain unconditionally to avoid cmake from downloading it.

Reply:
Ah, not really. But it can't hurt to set this unconditionally anyways.

Reply:
As of 4.6.1, this one is in an `if()` that would otherwise be an unused variable:
https://github.com/ccache/ccache/blob/v4.6.1/CMakeLists.txt#L103

This changed in master though:
https://github.com/ccache/ccache/blob/master/CMakeLists.txt#L115-L116